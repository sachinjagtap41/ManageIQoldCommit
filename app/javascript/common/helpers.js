// todo: expose this from patternfly-react instead?
export const getDisplayName = Component =>
  Component.displayName || Component.name || 'Component';
