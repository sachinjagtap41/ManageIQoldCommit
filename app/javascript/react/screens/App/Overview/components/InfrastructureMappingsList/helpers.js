export const mapInfrastructureMappings = (
  transformation_mapping_items,
  clusters,
  datastores,
  networks
) => {
  /**
   * map the target source -> destination clusters/networks/datastores for
   * display on the infrastructure mappings list view
   *
   * SQL-JOIN-MIQ-IN-JS ;)
   *
   * results in the following structure:
   * {
   *  targetClusters: {
   *    `clusterId`: {
   *      targetCluster: {...},
   *      sourceClusters: [{...}]
   *    }
   *  },
   *  targetDatastores: {
   *   `datastoreId`: {
   *     target: {
   *       targetCluster: {...},
   *       targetDatastore: {...}
   *     },
   *     sources: [
   *       {
   *         sourceDatastore: {...},
   *         sourceCluster: {...}
   *       }
   *     ]
   *   }
   *  }
   *  targetNetworks: {
   *   `networkId`: {
   *     target: {
   *       targetCluster: {...},
   *       targetNetwork: {...}
   *     },
   *     sources: [
   *       {
   *         sourceNetwork: {...},
   *         sourceCluster: {...}
   *       }
   *     ]
   *   }
   *  }
   * }
   *
   */
  const clusterMappingItems = transformation_mapping_items.filter(
    item => item.destination_type.toLowerCase() === 'emscluster'
  );
  const datastoreMappingItems = transformation_mapping_items.filter(
    item => item.destination_type.toLowerCase() === 'storage'
  );
  const networkMappingItems = transformation_mapping_items.filter(
    item => item.destination_type.toLowerCase() === 'lan'
  );

  // transform cluster mappings to key/value
  const clustersByMappingId = {};
  clusterMappingItems.forEach(cluster => {
    clustersByMappingId[cluster.transformation_mapping_id] = cluster;
  });

  // create unique cluster mappings by unique target cluster
  const targetClusters = {};
  clusterMappingItems.forEach(clusterMapping => {
    const sourceCluster = clusters.find(c => c.id === clusterMapping.source_id);
    const targetCluster = clusters.find(
      c => c.id === clusterMapping.destination_id
    );
    if (sourceCluster && targetCluster) {
      if (targetClusters[targetCluster.id]) {
        targetClusters[targetCluster.id].sourceClusters.push(sourceCluster);
      } else {
        targetClusters[targetCluster.id] = {};
        targetClusters[targetCluster.id].targetCluster = targetCluster;
        targetClusters[targetCluster.id].sourceClusters = [sourceCluster];
      }
    }
  });

  // create unique datastore mappings by unique target datastore
  const targetDatastores = {};
  datastoreMappingItems.forEach(datastoreMapping => {
    const clusterMapping =
      clustersByMappingId[datastoreMapping.transformation_mapping_id];
    const sourceCluster = clusters.find(c => c.id === clusterMapping.source_id);
    const targetCluster = clusters.find(
      c => c.id === clusterMapping.destination_id
    );
    const sourceDatastore = datastores.find(
      d => d.id === datastoreMapping.source_id
    );
    const targetDatastore = datastores.find(
      d => d.id === datastoreMapping.destination_id
    );

    if (
      clusterMapping &&
      sourceCluster &&
      targetCluster &&
      sourceDatastore &&
      targetDatastore
    ) {
      const source = {
        sourceDatastore,
        sourceCluster
      };

      const target = {
        targetDatastore,
        targetCluster
      };
      if (targetDatastores[targetDatastore.id]) {
        targetDatastores[targetDatastore.id].sources.push(source);
      } else {
        targetDatastores[targetDatastore.id] = {};
        targetDatastores[targetDatastore.id].target = target;
        targetDatastores[targetDatastore.id].sources = [source];
      }
    }
  });

  // create unique networks mappings by unique target network (using ems_uid)
  const targetNetworks = {};
  networkMappingItems.forEach(networkMapping => {
    const clusterMapping =
      clustersByMappingId[networkMapping.transformation_mapping_id];
    const sourceCluster = clusters.find(c => c.id === clusterMapping.source_id);
    const targetCluster = clusters.find(
      c => c.id === clusterMapping.destination_id
    );
    const sourceNetwork = networks.find(d => d.id === networkMapping.source_id);
    const targetNetwork = networks.find(
      d => d.id === networkMapping.destination_id
    );

    if (
      clusterMapping &&
      sourceCluster &&
      targetCluster &&
      sourceNetwork &&
      targetNetwork
    ) {
      const source = {
        sourceNetwork,
        sourceCluster
      };

      const target = {
        targetNetwork,
        targetCluster
      };
      // LANs are currently duplicated in the backend database model, so
      // we dedupe them using uid_ems attribute for now.
      if (targetNetworks[targetNetwork.uid_ems]) {
        const duplicatedLanIndex = targetNetworks[
          targetNetwork.uid_ems
        ].sources.findIndex(
          s => s.sourceNetwork.uid_ems === sourceNetwork.uid_ems
        );
        if (duplicatedLanIndex === -1) {
          targetNetworks[targetNetwork.uid_ems].sources.push(source);
        }
      } else {
        targetNetworks[targetNetwork.uid_ems] = {};
        targetNetworks[targetNetwork.uid_ems].target = target;
        targetNetworks[targetNetwork.uid_ems].sources = [source];
      }
    }
  });

  return {
    targetClusters,
    targetDatastores,
    targetNetworks
  };
};
