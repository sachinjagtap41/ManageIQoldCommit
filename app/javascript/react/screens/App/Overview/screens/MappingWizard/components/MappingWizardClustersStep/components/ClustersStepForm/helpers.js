export const targetClusterWithExtendedData = targetCluster => ({
  ...targetCluster,
  text: targetCluster.name,
  state: {
    expanded: true
  },
  selectable: true,
  selected: false
});

export const sourceClusterWithExtendedData = sourceCluster => ({
  ...sourceCluster,
  text: sourceCluster.name,
  icon: 'fa fa-file-o'
});

export const updateMapping = (
  clustersStepMapping,
  targetClusterToAddTo,
  sourceClustersToAdd
) => {
  const { nodes: sourceClusters, ...targetCluster } = clustersStepMapping;

  if (targetCluster.id === targetClusterToAddTo.id) {
    return {
      ...targetCluster,
      nodes: [
        ...sourceClusters,
        ...sourceClustersToAdd.map(sourceCluster =>
          sourceClusterWithExtendedData(sourceCluster)
        )
      ]
    };
  }
  return clustersStepMapping;
};

export const createNewMapping = (targetCluster, sourceClusters) => ({
  ...targetClusterWithExtendedData(targetCluster),
  nodes: sourceClusters.map(sourceCluster =>
    sourceClusterWithExtendedData(sourceCluster)
  )
});
