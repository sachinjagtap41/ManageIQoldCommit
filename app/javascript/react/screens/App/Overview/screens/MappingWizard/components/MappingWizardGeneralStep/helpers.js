export const asyncValidate = (values, dispatch, props) =>
  new Promise((resolve, reject) => {
    const { name: newTransformationName } = values;
    const { transformationMappings } = props;
    const existingTransformationNames = transformationMappings.reduce(
      (names, transformationMapping) => [
        ...names,
        transformationMapping.name.trim()
      ],
      []
    );
    const duplicateName = existingTransformationNames.find(
      existingTransformationName =>
        existingTransformationName === newTransformationName.trim()
    );
    if (duplicateName) {
      props.showAlertAction(
        sprintf(
          __('Infrastructure mapping %s already exists'),
          newTransformationName
        )
      );
      const error = { name: __('Please enter a unique name') };
      reject(error);
    } else {
      resolve();
    }
  });

export const onChange = (values, dispatch, props) => {
  if (props.valid) {
    props.hideAlertAction();
  }
};
