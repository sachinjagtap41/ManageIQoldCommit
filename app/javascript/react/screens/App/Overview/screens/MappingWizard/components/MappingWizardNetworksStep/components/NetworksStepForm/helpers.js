import { groupByUidEms } from '../../MappingWizardNetworksStepSelectors';

export const getRepresentatives = (groupedNetworks = {}) => {
  const representatives = [];
  for (const uid_ems in groupedNetworks) {
    if ({}.hasOwnProperty.call(groupedNetworks, uid_ems)) {
      representatives.push(groupedNetworks[uid_ems][0]);
    }
  }
  return representatives;
};

export const sourceNetworksFilter = (
  groupedSourceNetworks,
  networksStepMappings
) => {
  const mappedNetworks = networksStepMappings.reduce(
    (mappedNetworksArray, targetClusterWithNetworkMappings) => {
      const sourceNetworks = targetClusterWithNetworkMappings.nodes.reduce(
        (networks, networkMapping) => [...networks, ...networkMapping.nodes],
        []
      );
      return [...mappedNetworksArray, ...sourceNetworks];
    },
    []
  );

  return getRepresentatives(groupedSourceNetworks).filter(
    network =>
      !mappedNetworks.some(
        mappedNetwork => mappedNetwork.uid_ems === network.uid_ems
      )
  );
};

export const clustersMappingWithTreeViewAttrs = clusterMapping => ({
  ...clusterMapping,
  text: clusterMapping.name,
  selectable: false
});

export const targetNetworkWithTreeViewAttrs = targetNetwork => ({
  ...targetNetwork,
  text: targetNetwork.name,
  selectable: true,
  selected: false,
  state: {
    expanded: true
  }
});

export const sourceNetworkWithTreeViewAttrs = (
  sourceNetwork,
  selectedCluster
) => ({
  ...sourceNetwork,
  text: sourceNetwork.name,
  icon: 'fa fa-file-o',
  sourceClusterId: selectedCluster.id,
  selectable: true,
  selected: false
});

export const networkGroupingForRep = (
  sourceNetworkRep,
  groupedSourceNetworks,
  selectedCluster
) => {
  const sourceNetworks = groupedSourceNetworks[sourceNetworkRep.uid_ems];
  return sourceNetworks.map(sourceNetwork =>
    sourceNetworkWithTreeViewAttrs(sourceNetwork, selectedCluster)
  );
};

export const dedupeMappedSourceNetworks = networksMapping => {
  const { nodes: sourceNetworks, ...targetNetwork } = networksMapping;
  const groupedNetworks = groupByUidEms(sourceNetworks);
  return {
    ...targetNetwork,
    nodes: getRepresentatives(groupedNetworks)
  };
};

export const dedupeMappedTargetNetworks = networksStepMapping => {
  const { nodes: networksMappings, ...targetCluster } = networksStepMapping;
  const groupedMappings = groupByUidEms(networksMappings);
  return {
    ...targetCluster,
    nodes: getRepresentatives(groupedMappings)
  };
};

export const mappingsForTreeView = mappings =>
  mappings.reduce((updatedMappings, networksStepMapping) => {
    const { nodes: networksMappings } = dedupeMappedTargetNetworks(
      networksStepMapping
    );
    const updatedNetworksMappings = networksMappings.reduce(
      (dedupedMappings, networksMapping) => [
        ...dedupedMappings,
        dedupeMappedSourceNetworks(networksMapping)
      ],
      []
    );
    const updatedMapping = {
      ...networksStepMapping,
      nodes: updatedNetworksMappings
    };
    return [...updatedMappings, updatedMapping];
  }, []);

export const mappingWithTargetNetworkRemoved = (
  networksStepMapping,
  targetNetworkToRemove
) => {
  const { nodes: networksMappings, ...targetCluster } = networksStepMapping;
  const updatedNetworksMappings = networksMappings.filter(
    targetNetworkWithSourceNetworks =>
      targetNetworkWithSourceNetworks.id !== targetNetworkToRemove.id
  );
  return updatedNetworksMappings.length === 0
    ? null
    : {
        ...targetCluster,
        nodes: updatedNetworksMappings
      };
};

export const mappingWithSourceNetworkRemoved = (
  networksMapping,
  sourceNetworkToRemove
) => {
  const { nodes: sourceNetworks, ...targetNetwork } = networksMapping;
  const updatedSourceNetworks = sourceNetworks.filter(
    sourceNetwork => sourceNetwork.uid_ems !== sourceNetworkToRemove.uid_ems
  );
  return updatedSourceNetworks.length === 0
    ? null
    : {
        ...targetNetwork,
        nodes: updatedSourceNetworks
      };
};
