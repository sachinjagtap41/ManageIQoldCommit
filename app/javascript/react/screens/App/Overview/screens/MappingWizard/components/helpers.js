export const getClusterOptions = clusterMappings => {
  const sourceClustersWithAssociatedTargetClusters = clusterMappings.reduce(
    (mappings, targetClusterWithSourceClusters) => {
      const {
        nodes: sourceClusters,
        ...targetCluster
      } = targetClusterWithSourceClusters;
      const sourceToTargetMappings = sourceClusters.map(sourceCluster => ({
        sourceCluster,
        targetCluster,
        sourceClusterMappedToTargetCluster: {
          name: `${sourceCluster.name} (${targetCluster.name})`,
          id: sourceCluster.id
        }
      }));
      return mappings.concat(sourceToTargetMappings);
    },
    []
  );

  return sourceClustersWithAssociatedTargetClusters.map(
    ({ sourceClusterMappedToTargetCluster }) =>
      sourceClusterMappedToTargetCluster
  );
};
