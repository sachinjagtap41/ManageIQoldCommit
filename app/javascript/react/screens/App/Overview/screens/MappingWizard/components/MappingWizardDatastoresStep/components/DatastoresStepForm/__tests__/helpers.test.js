import { removeSourceDatastore, updateMappings } from '../helpers';

import {
  targetClusters,
  sourceDatastores,
  targetDatastores
} from '../DatastoresStepForm.fixtures';

describe('removeSourceDatastore', () => {
  let targetDatastore;

  beforeEach(() => {
    [targetDatastore] = targetDatastores;
  });

  test('removes specified source datastore', () => {
    const [
      sourceDatastoreToRemove,
      sourceDatastoreShouldRemain
    ] = sourceDatastores;
    const datastoresMapping = {
      ...targetDatastore,
      nodes: [sourceDatastoreToRemove, sourceDatastoreShouldRemain]
    };

    expect(
      removeSourceDatastore(datastoresMapping, sourceDatastoreToRemove)
    ).toMatchSnapshot();
  });

  test('removes entire mapping if no source datastores remain', () => {
    const [sourceDatastoreToRemove] = sourceDatastores;
    const datastoresMapping = {
      ...targetDatastore,
      nodes: [sourceDatastoreToRemove]
    };

    expect(
      removeSourceDatastore(datastoresMapping, sourceDatastoreToRemove)
    ).toMatchSnapshot();
  });
});

describe('updateMappings', () => {
  let targetCluster;

  beforeEach(() => {
    [targetCluster] = targetClusters;
  });

  test('removes specified source datastore', () => {
    const [
      sourceDatastoreToRemove,
      sourceDatastoreShouldRemain
    ] = sourceDatastores;
    const [targetDatastore] = targetDatastores;
    const datastoresStepMapping = {
      ...targetCluster,
      nodes: [
        {
          ...targetDatastore,
          nodes: [sourceDatastoreToRemove, sourceDatastoreShouldRemain]
        }
      ]
    };

    expect(
      updateMappings(datastoresStepMapping, sourceDatastoreToRemove)
    ).toMatchSnapshot();
  });

  describe('selected node is a target datastore', () => {
    test('removes specified target datastore along with its mapped source datastores', () => {
      const [
        targetDatastoreToRemove,
        targetDatastoreShouldRemain
      ] = targetDatastores;
      const [
        sourceDatastoreToRemove,
        sourceDatastoreShouldRemain
      ] = sourceDatastores;
      const datastoresStepMapping = {
        ...targetCluster,
        nodes: [
          {
            ...targetDatastoreToRemove,
            nodes: [sourceDatastoreToRemove]
          },
          {
            ...targetDatastoreShouldRemain,
            nodes: [sourceDatastoreShouldRemain]
          }
        ]
      };

      expect(
        updateMappings(datastoresStepMapping, targetDatastoreToRemove)
      ).toMatchSnapshot();
    });

    test('removes entire datastores step mapping if no target datastores remain', () => {
      const [targetDatastoreToRemove] = targetDatastores;
      const [sourceDatastoreToRemove] = sourceDatastores;
      const datastoresStepMapping = {
        ...targetCluster,
        nodes: [
          {
            ...targetDatastoreToRemove,
            nodes: [sourceDatastoreToRemove]
          }
        ]
      };

      expect(
        updateMappings(datastoresStepMapping, targetDatastoreToRemove)
      ).toMatchSnapshot();
    });
  });
});
