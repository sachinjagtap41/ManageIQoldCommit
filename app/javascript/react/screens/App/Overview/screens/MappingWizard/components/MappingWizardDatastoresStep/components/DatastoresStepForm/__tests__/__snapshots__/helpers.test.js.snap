// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeSourceDatastore removes entire mapping if no source datastores remain 1`] = `undefined`;

exports[`removeSourceDatastore removes specified source datastore 1`] = `
Object {
  "created_on": "2017-10-06T16:41:09Z",
  "directory_hierarchy_supported": true,
  "ems_ref": "datastore-874",
  "ems_ref_obj": "datastore-874",
  "free_space": 3389349027840,
  "href": "http://localhost:3000/api/data_stores/13",
  "id": "1-1",
  "last_perf_capture_on": "2018-01-21T01:00:00Z",
  "last_scan_on": "2018-01-21T00:05:45Z",
  "location": "a0f0178e-ddec1f02-0000-000000000000",
  "master": false,
  "multiplehostaccess": 1,
  "name": "RHVCluster1-Shared-NFS-1",
  "nodes": Array [
    Object {
      "created_on": "2017-10-06T16:41:09Z",
      "directory_hierarchy_supported": true,
      "ems_ref": "datastore-871",
      "ems_ref_obj": "datastore-871",
      "free_space": 1109023260672,
      "href": "http://localhost:3000/api/data_stores/2",
      "id": "1-2",
      "last_perf_capture_on": "2018-01-21T01:00:00Z",
      "last_scan_on": "2018-01-21T00:00:56Z",
      "location": "3ac00f25-19a416be-56d6-000e1eaac6a0",
      "master": false,
      "multiplehostaccess": 0,
      "name": "VMWareCluster1-DS2",
      "raw_disk_mappings_supported": true,
      "storage_domain_type": null,
      "store_type": "VMFS",
      "thin_provisioning_supported": true,
      "total_space": 1192121860096,
      "uncommitted": 63288031245,
      "updated_on": "2018-01-21T01:02:35Z",
    },
  ],
  "raw_disk_mappings_supported": false,
  "storage_domain_type": null,
  "store_type": "NFS41",
  "thin_provisioning_supported": true,
  "total_space": 5388852264960,
  "uncommitted": 3917934592000,
  "updated_on": "2018-01-21T01:02:38Z",
}
`;

exports[`updateMappings removes specified source datastore 1`] = `
Object {
  "created_on": "2017-10-06T16:41:09Z",
  "drs_automation_level": "fullyAutomated",
  "drs_enabled": false,
  "drs_migration_threshold": 3,
  "effective_cpu": 109356,
  "effective_memory": 790163881984,
  "ems_id": "1",
  "ems_ref": "domain-c7",
  "ems_ref_obj": "domain-c7",
  "ha_admit_control": true,
  "ha_enabled": false,
  "ha_max_failures": 1,
  "href": "http://localhost:3000/api/clusters/1",
  "id": "1",
  "last_perf_capture_on": null,
  "name": "RHV-1",
  "nodes": Array [
    Object {
      "created_on": "2017-10-06T16:41:09Z",
      "directory_hierarchy_supported": true,
      "ems_ref": "datastore-874",
      "ems_ref_obj": "datastore-874",
      "free_space": 3389349027840,
      "href": "http://localhost:3000/api/data_stores/13",
      "id": "1-1",
      "last_perf_capture_on": "2018-01-21T01:00:00Z",
      "last_scan_on": "2018-01-21T00:05:45Z",
      "location": "a0f0178e-ddec1f02-0000-000000000000",
      "master": false,
      "multiplehostaccess": 1,
      "name": "RHVCluster1-Shared-NFS-1",
      "nodes": Array [
        Object {
          "created_on": "2017-10-06T16:41:09Z",
          "directory_hierarchy_supported": true,
          "ems_ref": "datastore-871",
          "ems_ref_obj": "datastore-871",
          "free_space": 1109023260672,
          "href": "http://localhost:3000/api/data_stores/2",
          "id": "1-2",
          "last_perf_capture_on": "2018-01-21T01:00:00Z",
          "last_scan_on": "2018-01-21T00:00:56Z",
          "location": "3ac00f25-19a416be-56d6-000e1eaac6a0",
          "master": false,
          "multiplehostaccess": 0,
          "name": "VMWareCluster1-DS2",
          "raw_disk_mappings_supported": true,
          "storage_domain_type": null,
          "store_type": "VMFS",
          "thin_provisioning_supported": true,
          "total_space": 1192121860096,
          "uncommitted": 63288031245,
          "updated_on": "2018-01-21T01:02:35Z",
        },
      ],
      "raw_disk_mappings_supported": false,
      "storage_domain_type": null,
      "store_type": "NFS41",
      "thin_provisioning_supported": true,
      "total_space": 5388852264960,
      "uncommitted": 3917934592000,
      "updated_on": "2018-01-21T01:02:38Z",
    },
  ],
  "type": null,
  "uid_ems": "domain-c7",
  "updated_on": "2018-01-20T23:12:00Z",
}
`;

exports[`updateMappings selected node is a target datastore removes entire datastores step mapping if no target datastores remain 1`] = `undefined`;

exports[`updateMappings selected node is a target datastore removes specified target datastore along with its mapped source datastores 1`] = `
Object {
  "created_on": "2017-10-06T16:41:09Z",
  "drs_automation_level": "fullyAutomated",
  "drs_enabled": false,
  "drs_migration_threshold": 3,
  "effective_cpu": 109356,
  "effective_memory": 790163881984,
  "ems_id": "1",
  "ems_ref": "domain-c7",
  "ems_ref_obj": "domain-c7",
  "ha_admit_control": true,
  "ha_enabled": false,
  "ha_max_failures": 1,
  "href": "http://localhost:3000/api/clusters/1",
  "id": "1",
  "last_perf_capture_on": null,
  "name": "RHV-1",
  "nodes": Array [
    Object {
      "created_on": "2017-10-06T16:41:09Z",
      "directory_hierarchy_supported": true,
      "ems_ref": "datastore-874",
      "ems_ref_obj": "datastore-874",
      "free_space": 3389349027840,
      "href": "http://localhost:3000/api/data_stores/15",
      "id": "2-1",
      "last_perf_capture_on": "2018-01-21T01:00:00Z",
      "last_scan_on": "2018-01-21T00:05:45Z",
      "location": "a0f0178e-ddec1f02-0000-000000000000",
      "master": false,
      "multiplehostaccess": 1,
      "name": "RHVCluster2-Shared-NFS-1",
      "nodes": Array [
        Object {
          "created_on": "2017-10-06T16:41:09Z",
          "directory_hierarchy_supported": true,
          "ems_ref": "datastore-871",
          "ems_ref_obj": "datastore-871",
          "free_space": 1109023260672,
          "href": "http://localhost:3000/api/data_stores/2",
          "id": "1-2",
          "last_perf_capture_on": "2018-01-21T01:00:00Z",
          "last_scan_on": "2018-01-21T00:00:56Z",
          "location": "3ac00f25-19a416be-56d6-000e1eaac6a0",
          "master": false,
          "multiplehostaccess": 0,
          "name": "VMWareCluster1-DS2",
          "raw_disk_mappings_supported": true,
          "storage_domain_type": null,
          "store_type": "VMFS",
          "thin_provisioning_supported": true,
          "total_space": 1192121860096,
          "uncommitted": 63288031245,
          "updated_on": "2018-01-21T01:02:35Z",
        },
      ],
      "raw_disk_mappings_supported": false,
      "storage_domain_type": null,
      "store_type": "NFS41",
      "thin_provisioning_supported": true,
      "total_space": 5388852264960,
      "uncommitted": 3917934592000,
      "updated_on": "2018-01-21T01:02:38Z",
    },
  ],
  "type": null,
  "uid_ems": "domain-c7",
  "updated_on": "2018-01-20T23:12:00Z",
}
`;
